### PR TITLE
Integrate container image signing in `krel stage`

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:
Pushed images and manifest lists on stage are now signed by using the release-sdk.
#### Which issue(s) this PR fixes:

Refers to #2383 
#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
Added container image signing for intermediate container images produced by `krel stage`
```

### CODE REVIEW

File: pkg/release/images.go [hunk 1]
```diff
@@ -78,6 +80,11 @@ func (*defaultCommandClient) RepoTagFromTarball(path string) (string, error) {
 	return tagOutput.OutputTrimNL(), nil
 }
 
+func (*defaultCommandClient) Sign(signer *sign.Signer, reference string) error {
```
Comment (saschagrunert) [inline pkg/release/images.go:83]: We could find a better name for this interface

File: pkg/release/images.go [hunk 2]
```diff
@@ -166,6 +179,10 @@ func (i *Images) Publish(registry, version, buildPath string) error {
 		); err != nil {
 			return errors.Wrap(err, "push manifest")
 		}
+
+		if err := i.client.Sign(signer, imageVersion); err != nil {
```
Comment (saschagrunert) [inline pkg/release/images.go:183]: `imageVersion` could be `manifest`

**Review by puerco — CHANGES_REQUESTED** [review]
Nice work @saschagrunert !
As we made some changes to the project hosting the signer accounts, please update the service accounts in these two:

**Comment by saschagrunert** [comment]
Switched to sigs.k8s.io/release-sdk v0.7.0 since this PR got a lgtm.

**Review by cpanato — APPROVED** [review]
/lgtm

**Comment by puerco** [comment]
Thank you @saschagrunert !!!
I've rebased your branch and [ran a test run of staging](https://console.cloud.google.com/cloud-build/builds;region=global/54598f84-6a29-4b4c-a92b-fd8d09f0258b?project=kubernetes-release-test) now that we have the permissions in place. I am happy to report that the staged images are signed! :tada::tada::tada::tada:

/lgtm
/approve

image

**Comment by saschagrunert** [comment]
Thanks for merging @puerco! :pray:

**Pull request merged**